### ISSUE TITLE
maptile_geohelp does not launch in command line mode

### ISSUE THREAD

**Issue by dalek2point3** [michaelstepner/maptile#11]
When running stata in command line mode maptile_geohelp does not launch

![image](https://cloud.githubusercontent.com/assets/6828439/17679961/afc58e14-62f2-11e6-98ba-9a011543bf6a.png)

**Comment by michaelstepner** [comment]
Thanks for the report.

I'm not sure how to fix it. The -view- command is what opens and renders smcl file, except apparently it doesn't work in Stata's console mode.  Some quick googling doesn't turn up any commands that do print an smcl file to the console window. Although Stata no doubt has this ability, because all Stata help files are written in SMCL, and `help regress` for example just prints the whole file to the console window.

Do you have any suggestions for how to display an SMCL file in Console Stata using existing Stata commands? (The alternative route would be writing an ado program that does so, which is definitely feasible.)

**Comment by dalek2point3** [comment]
I'm not sure. It's strange because "help maptile" works just fine. Its just that the geography help files dont show up. Is the code doing anything different for the geography files vs. the maptile file?

**Comment by michaelstepner** [comment]
Yes, the code for `help` and `maptile_geohelp` is different, even though the format of the file being opened is the same.  And programming `maptile_geohelp` to do the same thing as `help` isn't an option.

Eventually, I will likely have to write a program that manually handles printing smcl files in Console Stata. In the meantime, as workarounds, I'd suggest:

1. Running maptile_geohelp in GUI Stata if that's available to you.
2. Instead of running `maptile_install using "http://files.michaelstepner.com/geo_<GEONAME>.zip"`, paste that URL into your browser and download the zip file to your computer. (Or use `wget` if you're on a computer without any GUI.) Unzip it, and open the SMCL file in a text editor. It won't look pretty, but you can read it.

**Comment by dalek2point3** [comment]
Thanks for the workaround! This is surely not a high-priority issue, but thought I would document it as a to-do for some time in the future ... I have access to gui, but I can see people  using this on servers who might run into problems.

**Comment by michaelstepner** [comment]
I've added it to my to-do list, as a "do this eventually when you're next working on maptile" kind of thing. Thanks again for reporting!

**Comment by michaelstepner** [comment]
@dalek2point3, I think I've fixed this issue in the [latest commit on the develop branch](https://github.com/michaelstepner/maptile/commit/98a43ea364bcc0fd5cec6d2284db5c2cbcbe14f4).

The solution came from a suggestion from @wbuchanan on [Statalist](http://www.statalist.org/forums/forum/general-stata-discussion/general/1353761-printing-rendered-smcl-file-in-console-stata).

@dalek2point3, I tested it out on my computer, but if you'd like to test the bug fix on your computer, you could run:

```
ssc uninstall maptile
net install maptile, from(https://github.com/michaelstepner/maptile/raw/develop)
```

I haven't updated the version of maptile in the SSC with this bug fix yet, but the command above will install maptile directly from the develop branch of the repo.

**Comment by michaelstepner** [comment]
This fix was merged into v1.0.5 of maptile in commit fb5deb1. It's now available on Stata's SSC.
